src/logicTemplates/periodicpayment.js

```diff
@@ -0,0 +1,101 @@
+        if (!Number.isSafeInteger(withdrawalWindow) || withdrawalWindow < 0) throw Error("withdrawalWindow must be a positive number and smaller than 2^53-1");
+        if (!Number.isSafeInteger(period) || period < 0) throw Error("period must be a positive number and smaller than 2^53-1");
+        if (!Number.isSafeInteger(expiryRound) || expiryRound < 0) throw Error("expiryRound must be a positive number and smaller than 2^53-1");
+        if (!Number.isSafeInteger(maxFee) || maxFee < 0) throw Error("maxFee must be a positive number and smaller than 2^53-1");
+
+        if (lease === undefined) {
+            let leaseBytes = nacl.randomBytes(32);
+            lease = Buffer.from(leaseBytes).toString('base64');
+        }
+        const referenceProgramB64 = "ASAHAQYFAAQDByYCIAECAwQFBgcIAQIDBAUGBwgBAgMEBQYHCAECAwQFBgcIIJKvkYTkEzwJf2arzJOxERsSogG9nQzKPkpIoc4TzPTFMRAiEjEBIw4QMQIkGCUSEDEEIQQxAggSEDEGKBIQMQkyAxIxBykSEDEIIQUSEDEJKRIxBzIDEhAxAiEGDRAxCCUSEBEQ";
+        let referenceProgramBytes = Buffer.from(referenceProgramB64, 'base64');
+        let referenceOffsets = [ /*fee*/ 4 /*period*/, 5 /*withdrawWindow*/, 7 /*amount*/, 8 /*expiryRound*/, 9 /*lease*/, 12 /*receiver*/, 46];
+        let injectionVector =  [maxFee, period, withdrawalWindow,
+                                amount, expiryRound, lease,
+                                receiver];
+        let injectionTypes = [templates.valTypes.INT, templates.valTypes.INT, templates.valTypes.INT,
+                                templates.valTypes.INT, templates.valTypes.INT, templates.valTypes.BASE64,
+                                templates.valTypes.ADDRESS];
+        let injectedBytes = templates.inject(referenceProgramBytes, referenceOffsets, injectionVector, injectionTypes);
+        this.programBytes = injectedBytes;
+        let lsig = new logicSig.LogicSig(injectedBytes, undefined);
+        this.address = lsig.address();
+        this.receiver = receiver;
+        this.amount = amount;
+        this.duration = withdrawalWindow;
+        this.period = period;
+    }
+
+    /**
+     * returns the program bytes
+     * @returns {Uint8Array}
+     */
+    getProgram() {
+        return this.programBytes;
+    }
+
+    /**
+     * returns the string address of the contract
+     * @returns {string}
+     */
+    getAddress() {
+        return this.address;
+    }
+
+    /**
+     * getWithdrawalTransaction returns a signed transaction extracting funds form the contract
+     * @param {int} fee: the fee to pay in microAlgos
+     * @param {int} firstValid: the first round on which the txn will be valid
+     * @param {string} genesisHash: the hash representing the network for the txn
+     * @returns {Object} Object containing txID and blob representing signed transaction
+     * @throws error on failure
+     */
+    getWithdrawalTransaction(fee, firstValid, genesisHash) {
```

Comment:
Do you expect the receiver to re-initiate the contract? Seems like you should be able to parse all the info from the txn and lsig received from the creator.